v2/internal/util/cel/json_type_provider.go

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package cel
+
+import (
+	"encoding/json"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+const (
+	typeName = "v1.JSON"
+)
+
+var jsonMapType = types.NewMapType(types.StringType, types.NewObjectType(typeName))
+
+// TODO: Tests
```

Comment:
Still pending?

Reply:
Not really, it's covered by the tests we have now. This was from when I originally wrote this and didn't have all the tests yet.